src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py

```diff
@@ -0,0 +1,192 @@
+
+        self.blob_container = "blobContainer"
+        self.storage_account = "storageAccount"
+        self.storage_account_resource_group = "storageAccountResourceGroup"
+        self.storage_account_subsciption = "storageAccountSubscriptionId"
+        self.cpu_limit = "cpuLimit"
+        self.memory_limit = "memoryLimit"
+
+        self.configuration_mapping = {
+            self.blob_container.lower(): self.BACKUP_STORAGE_ACCOUNT_CONTAINER,
+            self.storage_account.lower(): self.BACKUP_STORAGE_ACCOUNT_NAME,
+            self.storage_account_resource_group.lower(): self.BACKUP_STORAGE_ACCOUNT_RESOURCE_GROUP,
+            self.storage_account_subsciption.lower(): self.BACKUP_STORAGE_ACCOUNT_SUBSCRIPTION,
+            self.cpu_limit.lower(): self.RESOURCE_LIMIT_CPU,
+            self.memory_limit.lower(): self.RESOURCE_LIMIT_MEMORY
+        }
+
+        self.bsl_configuration_settings = [
+            self.blob_container,
+            self.storage_account,
+            self.storage_account_resource_group,
+            self.storage_account_subsciption
+        ]
+
+    def Create(
+        self,
+        cmd,
+        client,
+        resource_group_name,
+        cluster_name,
+        name,
+        cluster_type,
+        cluster_rp,
+        extension_type,
+        scope,
+        auto_upgrade_minor_version,
+        release_train,
+        version,
+        target_namespace,
+        release_namespace,
+        configuration_settings,
+        configuration_protected_settings,
+        configuration_settings_file,
+        configuration_protected_settings_file
+    ):
+        # Current scope of DataProtection Kubernetes Backup extension is 'cluster' #TODO: add TSGs when they are in place
+        if scope == 'namespace':
+            raise InvalidArgumentValueError(f"Invalid scope '{scope}'. This extension can only be installed at 'cluster' scope.")
+
+        scope_cluster = ScopeCluster(release_namespace=release_namespace)
+        ext_scope = Scope(cluster=scope_cluster, namespace=None)
+
+        if cluster_type.lower() != 'managedclusters':
+            raise InvalidArgumentValueError(f"Invalid cluster type '{cluster_type}'. This extension can only be installed for managed clusters.")
+
+        if release_namespace is not None:
+            logger.warning(f"Ignoring 'release-namespace': {release_namespace}")
+
+        tenant_id = self.__get_tenant_id(cmd.cli_ctx)
+        if not tenant_id:
+            raise SystemExit(logger.error("Unable to fetch TenantId. Please check your subscription or run 'az login' to login to Azure."))
+
+        self.__validate_and_map_config(configuration_settings)
+        self.__validate_backup_storage_account(cmd.cli_ctx, resource_group_name, cluster_name, configuration_settings)
+
+        configuration_settings[self.TENANT_ID] = tenant_id
+
+        if release_train is None:
+            release_train = 'stable'
+
+        create_identity = True
```

Comment:
I believe we can set this flag to false as identities are not created for managedClusters and if that is the only clusterType that will be supported and if it is not about to change.

[https://github.com/AzureArcForKubernetes/azure-cli-extensions/blob/k8s-extension/public/src/k8s-extension/azext_k8s_extension/custom.py#L416](https://github.com/AzureArcForKubernetes/azure-cli-extensions/blob/k8s-extension/public/src/k8s-extension/azext_k8s_extension/custom.py#L416)

Reply:
nvm, since we already have the logic present for AKS

Reply:
Got it Bavneet!